db/upgrade.php

```diff
@@ -230,5 +230,17 @@ function xmldb_moodleoverflow_upgrade($oldversion) {
         upgrade_mod_savepoint(true, 2022072000, 'moodleoverflow');
     }
 
+    if ($oldversion < 2022110700) {
+
+        $roleids = $DB->get_records_menu('role', null, '', 'shortname, id');
+        unassign_capability('mod/moodleoverflow:reviewpost', $roleids['manager']);
+        assign_capability(
+            'mod/moodleoverflow:reviewpost', CAP_ALLOW, $roleids['teacher'], context_system::instance()
```

Comment:
Why is this assigned again? The teacher capability was not changed. 
Additionally, there should be confirmations whether a manager or teacher role with this name exists, otherwise this will throw an exception.

Reply:
I did this because teachers do not have the capability 'moodle/course:manageactivities', only editingteachers do. I cloned the permissions from there without knowing that, so I wanted to give nonediting teachers the capability back.

Maybe I should look into whether it is easier to just remove the capability completely, and let moodle add it again, but this time with no clonepermissions and the correct archetype permissions?

Reply:
I have now looked into it and found a method which gives me all roles based on a archetype, so i've used that.